src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs

```diff
@@ -228,13 +228,24 @@ public override void ExecuteCmdlet()
                         Path = "/oauth2/token"
                     };
 
+                    var envSecret = System.Environment.GetEnvironmentVariable(MSISecretVariable);
+
                     var msiSecret = this.IsBound(nameof(ManagedServiceSecret))
                         ? ManagedServiceSecret.ConvertToString()
-                        : System.Environment.GetEnvironmentVariable(MSISecretVariable);
+                        : envSecret;
+
+                    var envUri = System.Environment.GetEnvironmentVariable(MSIEndpointVariable);
 
                     var suppliedUri = this.IsBound(nameof(ManagedServiceHostName))
                         ? builder.Uri.ToString()
-                        : System.Environment.GetEnvironmentVariable(MSIEndpointVariable);
+                        : envUri;
+
+                    if (!this.IsBound(nameof(ManagedServiceHostName)) && !string.IsNullOrWhiteSpace(envUri) 
```

Comment:
Do you think we should have an optional AppService swith parameter so the user can specify using the AppService style of MSI, (or an explicit ApiVersion parameter), so that the user cna opt into this even if secret and endpoint are specified?

Reply:
Since this is a short term fix I don't think we should change the cmdlet interface.